hashedindex/textparser.py

```diff
@@ -83,6 +87,9 @@ def word_tokenize(text, stopwords=_stopwords, ngrams=None, min_length=0, ignore_
 
     Generated tokens are lowercased by default; the retain_casing flag can
     be set to True to retain upper/lower casing from the original text.
+
+    When retain_punctuation is set to False (default), punctuation characters
+    are removed from the input text.
```

Comment:
This is slightly hard to read - maybe take a similar format for what you wrote for `retain_casing`?